### PR TITLE
add comment and answer comment

### PR DESCRIPTION
**Pull request by S-Toky**
Here is the branch for the comment and the answer of each comment

### CODE REVIEW

File: app/assets/javascripts/answers.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (Israel-Laguan) [inline app/assets/javascripts/answers.coffee]: this file is not needed, please erase

File: app/assets/stylesheets/comments.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the comments controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (Israel-Laguan) [inline app/assets/stylesheets/comments.scss]: this file is not needed, please erase

File: app/helpers/answers_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module AnswersHelper
+end
```
Comment (Israel-Laguan) [inline app/helpers/answers_helper.rb]: if not needed please erase file

File: app/helpers/comments_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
+end
```
Comment (Israel-Laguan) [inline app/helpers/comments_helper.rb]: if not needed please erase file

File: app/views/answers/create.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (Israel-Laguan) [inline app/views/answers/create.html.erb]: if not needed please erase file

File: app/views/answers/destroy.html.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (Israel-Laguan) [inline app/views/answers/destroy.html.erb]: if not needed please erase file

**Comment by Israel-Laguan** [comment]
image
 
image

If not needed, please erase files

File: app/models/comment.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :event
+
+  has_many :answers, dependent: :destroy 
+
+  validates :body, presence: true, allow_blank: false
+end
```
Comment (Israel-Laguan) [inline app/models/comment.rb]: Add EOF please

File: app/views/events/show.html.erb [hunk 8]
```diff
@@ -36,3 +36,36 @@
...
+<div>
+  <% @comments.each do |comment| %>
+    [<%= comment.created_at.strftime("%I:%M %p; %d-%m-%y")%>] 
+    <%= comment.user.name%> : 
+    <%= comment.body %>
+      <% if @current_user && @current_user == comment.user %>
+        <%= button_to 'Destroy', event_comment_path(@event, comment), method: :delete, class: 'btn btn-danger'%>
+      <% end %>
+
+      <%= form_with model: [comment, @answer] do |d| %>
```
Comment (Israel-Laguan) [inline app/views/events/show.html.erb]: Please change `d` for something more understandable

File: app/controllers/answers_controller.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+  private
+
+  def answer_params
+    params.require(:answer).permit(:body, :comment_id)
+  end
+  
+
+  def set_comment
+    @comment = Comment.find(params[:comment_id])
```
Comment (Israel-Laguan) [inline app/controllers/answers_controller.rb:26]: Add a guard if the comment is null

File: app/controllers/comments_controller.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:body, :event_id)
+  end
+
+  def set_event
+    @event = Event.find(params[:event_id])
```
Comment (Israel-Laguan) [inline app/controllers/comments_controller.rb:27]: Add a guard if the event is null

File: app/controllers/events_controller.rb [hunk 11]
```diff
@@ -12,6 +12,9 @@ def index
   # GET /events/1
   def show
     @current_user = current_user
+    @comment = Comment.new
+    @comments = @event.comments.order("created_at DESC")
+    @answer = Answer.new
```
Comment (Israel-Laguan) [inline app/controllers/events_controller.rb:18]: Add a guard or fallback error here if any of Comment, Event or Answer are null

File: app/views/events/index.html.erb [hunk 12]
```diff
@@ -25,6 +25,7 @@
         <td><%= link_to 'Show', event %></td>
         <td><%= link_to 'Edit', edit_event_path(event) %></td>
         <td><%= link_to 'Attend', attend_path(event) %></td>
+        <td><p><%= event.comments.count %> <%= (event.comments.count) == 1 ? 'Comment' : 'Comments'%></p></td>
```
Comment (Israel-Laguan) [inline app/views/events/index.html.erb:54]: What about its 0 or `<1`?

File: app/views/events/show.html.erb [hunk 13]
```diff
@@ -36,3 +36,36 @@
 
 <%= link_to 'Edit', edit_event_path(@event) %> |
 <%= link_to 'Back', events_path %>
+
+<%= form_with model: [@event, @comment] do |f| %>
+  <%= f.text_field :body %>
+  <%= f.submit "comment", disable_with: 'commenting...' %>
+<% end %>
```
Comment (Israel-Laguan) [inline app/views/events/show.html.erb]: Please separate this logic as is not for "show events", a partial is ok

File: app/views/events/show.html.erb (hunk 8 above)
Comment (Israel-Laguan) [inline app/views/events/show.html.erb]: Please move this form to another partial, as is not "show events" logic

File: config/routes.rb [hunk 14]
```diff
@@ -12,5 +14,12 @@
...
   get 'attend/:id' => 'events#attend_event', as: :attend
   resources :events
+  resources :events do
+    resources :comments, only: [:create, :destroy]
+  end
+
+  resources :comments do
+    resources :answers
+  end
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (Israel-Laguan) [inline config/routes.rb:24]: please erase this line

**Review by Israel-Laguan — CHANGES_REQUESTED** [review]
Just minor tweeks

File: app/assets/javascripts/answers.coffee (hunk 1 above)
Comment (S-Toky) [reply app/assets/javascripts/answers.coffee]: I have finished making the change

File: app/assets/stylesheets/comments.scss (hunk 2 above)
Comment (S-Toky) [reply app/assets/stylesheets/comments.scss]: I have finished making the change

File: app/helpers/answers_helper.rb (hunk 3 above)
Comment (S-Toky) [reply app/helpers/answers_helper.rb]: I have finished making the change

File: app/helpers/comments_helper.rb (hunk 4 above)
Comment (S-Toky) [reply app/helpers/comments_helper.rb]: I have finished making the change

File: app/views/answers/create.html.erb (hunk 5 above)
Comment (S-Toky) [reply app/views/answers/create.html.erb]: I have finished making the change

File: app/views/answers/destroy.html.erb (hunk 6 above)
Comment (S-Toky) [reply app/views/answers/destroy.html.erb]: I have finished making the change

File: app/models/comment.rb (hunk 7 above)
Comment (S-Toky) [reply app/models/comment.rb]: I don't understand? there is already the end?

File: app/views/events/show.html.erb (hunk 8 above)
Comment (S-Toky) [reply app/views/events/show.html.erb]: I have finished making the change

File: app/views/events/show.html.erb (hunk 13 above)
Comment (S-Toky) [reply app/views/events/show.html.erb]: I have finished making the change

File: config/routes.rb (hunk 14 above)
Comment (S-Toky) [reply config/routes.rb:24]: I have finished making the change

File: app/views/events/show.html.erb (hunk 8 above)
Comment (S-Toky) [reply app/views/events/show.html.erb]: I have finished making the change

File: app/views/events/index.html.erb (hunk 12 above)
Comment (S-Toky) [reply app/views/events/index.html.erb:54]: Pour 0, le comment prend un "s" mais c'est pour le 1 qu'il n'y a pas de "s".

File: app/controllers/events_controller.rb (hunk 11 above)
Comment (S-Toky) [reply app/controllers/events_controller.rb:18]: For security, I already put this line of code on each model comment and answer : "validates :body, presence: true, allow_blank: false" so the comment will not be created if it is empty. 
For the null, an answer depends on a comment and a comment depends on an event, so if the event is null, a comment can't exist, no way to create one. and with the event, in the create method, there is already a condition.

File: app/models/comment.rb (hunk 7 above)
Comment (Israel-Laguan) [reply app/models/comment.rb]: After the final `end` there should be a line, please check this example:
<pre>
```ruby
end

```
</pre>
Note that there is an "empty line"

File: app/controllers/events_controller.rb (hunk 11 above)
Comment (Israel-Laguan) [reply app/controllers/events_controller.rb:18]: Got it, so seems is ok

File: app/views/events/index.html.erb (hunk 12 above)
Comment (Israel-Laguan) [reply app/views/events/index.html.erb:54]: If there is no comment, it will say "0 Comments"?

File: app/models/comment.rb (hunk 7 above)
Comment (S-Toky) [reply app/models/comment.rb]: ok, i verify it and push today

File: app/views/events/index.html.erb (hunk 12 above)
Comment (S-Toky) [reply app/views/events/index.html.erb:54]: yes

File: app/controllers/comments_controller.rb (hunk 10 above)
Comment (S-Toky) [reply app/controllers/comments_controller.rb:27]: For security, I already put this line of code on each model comment and answer : "validates :body, presence: true, allow_blank: false" so the comment will not be created if it is empty.
For the null, an answer depends on a comment and a comment depends on an event, so if the event is null, a comment can't exist, no way to create one. and with the event, in the create method, there is already a condition.

File: app/controllers/answers_controller.rb (hunk 9 above)
Comment (S-Toky) [reply app/controllers/answers_controller.rb:26]: For security, I already put this line of code on each model comment and answer : "validates :body, presence: true, allow_blank: false" so the comment will not be created if it is empty.
For the null, an answer depends on a comment and a comment depends on an event, so if the event is null, a comment can't exist, no way to create one. and with the event, in the create method, there is already a condition.

**Comment by S-Toky** [comment]
conflicts resolved

File: app/controllers/answers_controller.rb (hunk 9 above)
Comment (Israel-Laguan) [reply app/controllers/answers_controller.rb:26]: got it, great!

File: app/controllers/comments_controller.rb (hunk 10 above)
Comment (Israel-Laguan) [reply app/controllers/comments_controller.rb:27]: got it

File: Gemfile [hunk 15]
```diff
@@ -63,3 +63,5 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+gem 'jquery-rails'
```
Comment (Israel-Laguan) [inline Gemfile:67]: This solves #57

**Review by Israel-Laguan — APPROVED** [review]

**Pull request merged**